### PR TITLE
blob export updated

### PR DESCRIPTION
**Pull request by snopoke**
* add missing doc types
* refactor a bit
* make management command easier to use
* only iterate through models in the target domain

@gcapalbo @emord

### CODE REVIEW

File: corehq/blobs/migrate.py [hunk 1]
```diff
@@ -360,6 +360,38 @@ def should_process(self, doc):
...
+                "Expected to find zip blob db backend (got %r)" % self.db)
+
+    def process_object(self, object):
+        blob_id = object.restore_blob_id
+        info = BlobInfo(identifier=blob_id, length=object.content_length, digest=None)
+        self.total_blobs += 1
+        db = get_blob_db()
+        try:
+            content = db.get(blob_id)
+        except NotFound as e:
```
Comment (dimagimon) [inline corehq/blobs/migrate.py]: F841 local variable 'e' is assigned to but never used

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/blobs/migrate.py, line 86 - E501 line too long (117 > 115 characters)
* corehq/blobs/migrate.py, line 337 - E501 line too long (116 > 115 characters)
* corehq/blobs/migrate.py, line 348 - E226 missing whitespace around arithmetic operator
* corehq/util/doc_processor/couch.py, line 48 - E501 line too long (120 > 115 characters)
* corehq/util/tests/test_doc_processor.py, line 81 - E501 line too long (118 > 115 characters)

File: corehq/blobs/migrate.py [hunk 2]
```diff
@@ -501,16 +513,20 @@ def migrate(self, filename=None, reset=False, max_retry=2, chunk_size=100):
...
-        xform.XFormError,
-        xform.SubmissionErrorLog,
-        ("HQSubmission", xform.XFormInstance),
-    ], SqlFormAttachmentExporter),
+    SqlModelMigrator(
+        "sql_xforms",
+        sql_xform.XFormAttachmentSQL,
+        SqlFormAttachmentExporter
+    ),
+    ExportByDomain("saved_exports", [
```
Comment (emord) [inline corehq/blobs/migrate.py]: I think I remember seeing this and restores and it looked like they were only the cached results? Or is this something different than the cache?

**Review by emord — COMMENTED** [review]
Looks good to me but there are merge conflicts

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/blobs/migrate.py, line 86 - E501 line too long (117 > 115 characters)
* corehq/blobs/migrate.py, line 337 - E501 line too long (116 > 115 characters)
* corehq/blobs/migrate.py, line 348 - E226 missing whitespace around arithmetic operator
* corehq/util/doc_processor/couch.py, line 48 - E501 line too long (120 > 115 characters)
* corehq/util/tests/test_doc_processor.py, line 81 - E501 line too long (118 > 115 characters)

File: corehq/blobs/migrate.py (hunk 2 above)
Comment (snopoke) [inline corehq/blobs/migrate.py]: yea, that's true but not sure what will happen if we restore the docs and the attachments are missing - guessing we'll have to manually remove the attachments info from the doc. Seems easy enough to just dump them

**Pull request merged**